src/Microsoft.AspNetCore.Razor.Evolution/Legacy/Tokenizer.cs

```diff
@@ -193,19 +193,15 @@ protected TSymbol Single(TSymbolType type)
 
         protected void StartSymbol()
         {
-            Buffer.Clear();
+            Debug.Assert(Buffer.Length == 0);
+            Debug.Assert(CurrentErrors.Count == 0);
```

Comment:
I moved clearing these to `EndSymbol`. It turns out that `CurrentStart` was never correct